anchor-spl has a version mismatch with spl_associated_token_account

### Issue

 When trying to build something that uses anchor-spl 0.25.0, I got the following error:

```rust
error[E0433]: failed to resolve: could not find `instruction` in `spl_associated_token_account`
 --> [REDACTED]/anchor-spl-0.25.0/src/associated_token.rs:9:44
  |
9 |     let ix = spl_associated_token_account::instruction::create_associated_token_account(
  |                                            ^^^^^^^^^^^ could not find `instruction` in `spl_associated_token_account`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `anchor-spl` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```

It appears that anchor-spl lists `spl-associated-token-account = 1.0.3` as minimum version but the structure of that crate changed from 1.0.3 to 1.0.5 (i.e. not using semantic versioning) in a breaking way.

### Solution
Change 
```
spl-associated-token-account = "1.0.3"
```
to
```
spl-associated-token-account = "1.0.5"
```
in the anchor-spl Cargo.toml

--- a/spl/Cargo.toml
+++ b/spl/Cargo.toml
@@ -23,5 +23,5 @@ anchor-lang = { path = "../lang", version = "0.25.0", features = ["derive"] }
 serum_dex = { git = "https://github.com/project-serum/serum-dex", rev = "1be91f2", version = "0.4.0", features = ["no-entrypoint"], optional = true }
 solana-program = "~1.10.29"
 spl-token = { version = "~3.3.0", features = ["no-entrypoint"], optional = true }
-spl-associated-token-account = { version = "~1.0.3", features = ["no-entrypoint"], optional = true }
+spl-associated-token-account = { version = "~1.0.5", features = ["no-entrypoint"], optional = true }
 mpl-token-metadata = { version = "1.2.7", optional = true, features = ["no-entrypoint"] }

